src/Common/src/Internal/Runtime/RuntimeConstants.cs

```diff
@@ -10,7 +10,12 @@ internal static class FatFunctionPointerConstants
         /// Offset by which fat function pointers are shifted to distinguish them
         /// from real function pointers.
         /// </summary>
+#if WASM
+        // WebAssembly uses index tables, not addresses for function pointers.  This is going to be contentious it is baked into Ilc and hence an Ilc for x64 will not be able to compile Wasm and vice versa.  Alternative ways round this?.
```

Comment:
Yeah, this scheme doesn't work anymore.

What I would suggest is to move this to the TargetDetails class:
* Delete the FatFunctionPointerConstants class here.
* Make a new TargetDetails.Aot.cs file and put a "FatFunctionPointerOffset" property in it. The file should go to src\Common\src\TypeSystem\Aot directory (doesn't exist yet). It will be similar to the existing TargetDetails.CodeGen.cs file. The reason why we want it as a partial class is because TargetDetails class is also used in the shipping runtime and fat function pointers don't exist there. The property would return the new value for WASM target architecture (TargetDetails is where we keep the architecture anyway).
* Add a reference to the new file to the ILCompiler.TypeSystem project.
* Redirect all uses of the constant within the compiler to the new property on TargetDetails (you can usually get TargetDetails if you have any type system entity, e.g. in the use within FatFunctionPointerNode class, you would access Method.Context.Target).
* For the reference in FunctionPointerOps.cs, just define the const locally and use `#if WASM` like you did here. That's part of the runtime libraries and those are arch specific by definition.